internal/controllers/core/cmd/controller.go

```diff
@@ -413,14 +431,20 @@ func (c *Controller) handleProbeResultFunc(ctx context.Context, name types.Names
 		}
 
 		ready := result == prober.Success || result == prober.Warning
-		c.updateStatus(name, func(status *CmdStatus) {
-			// TODO(milas): this isn't quite right - we might end up setting
-			// 	a terminated process to ready, for example; in practice, we
-			// 	should update internal state on any goroutine/async trackers
-			// 	and trigger a reconciliation, which can then evaluate the full
-			// 	state + current spec
+
+		// TODO(milas): this isn't quite right - we might end up setting
+		// 	a terminated process to ready, for example; in practice, we
+		// 	should update internal state on any goroutine/async trackers
+		// 	and trigger a reconciliation, which can then evaluate the full
+		// 	state + current spec
```

Comment:
I think we can resolve this now!

1. Check that `status.Running != nil` before setting `Ready` here
2. In `processStatuses`, for the `Running` branch, flip the `mutateStatus` and `initProbeWorker.Do` order call (so running is populated before probe starts so that the check here will work)

Reply:
(I can also do it in a PR later if you want to limit scope here)

Reply:
let's do this in a separate PR, since i think this is going to be more controversial